tests/init_cfg_with_error_handler.rs

```diff
@@ -0,0 +1,23 @@
+#[test]
+#[cfg(all(feature = "config_parsing", feature = "yaml_format"))]
+fn test_cfg_with_error_handler() {
+    use std::{
+        io::{self, Write},
+        path::Path,
+    };
+
+    let cfg = log4rs::config::load_config_file(
+        Path::new("./test_cfgs/test.yml"),
+        log4rs::config::Deserializers::default(),
+    );
+    assert!(cfg.is_ok());
+    let cfg = cfg.unwrap();
+
+    let res = log4rs::config::init_config_with_err_handler(
+        cfg,
+        Box::new(|e: &anyhow::Error| {
+            let _ = writeln!(io::stderr(), "log4rs: {}", e);
+        }),
```

Comment:
we should probably set some bool and then verify it gets set in order to verify this handler actually gets called.

Reply:
So I don't think it was actually getting called when I triggered an error (try writing the log to a directory). Either way, I pulled this test out for this PR. I'm having trouble working through a good provable and repeatable test. I wanted to pass in a buffer and read the contents like we do else where but anything that gets moved into the closure is gone so I'm stilling thinking about a better solution